### PR TITLE
DASHBASE-71 Avoid zero value is incorrectly interpreted by the `default` function

### PR DESCRIPTION
**Pull request by cuichenli**
When the value provided to the `default` function is `0`, the `default` function will be invoked and replace `0` with the default value
Reference: helm/helm#3164

### CODE REVIEW

File: dashbase/tests/test_0_replicas/service/searcher.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+kind: Service
```
Comment (sincejune) [inline dashbase/tests/test_0_replicas/service/searcher.yaml:1]: We can remove this file as it's useless in this testing.

**Review by sincejune — APPROVED** [review]
Good First PR!

File: dashbase/templates/_helpers.tpl [hunk 2]
```diff
@@ -188,7 +188,7 @@ spec:
       maxUnavailable: 0%
     type: RollingUpdate
   {{- end }}
-  replicas: {{ .replicas | default 1 }}
+  replicas: {{ if kindIs "float64" .replicas }}{{ .replicas }}{{ else }}{{ 1 }}{{ end }}
```
Comment (sincejune) [inline dashbase/templates/_helpers.tpl:191]: Probably we can directly use `1` instead of `{{ 1 }}`.

Comment (cuichenli) [reply dashbase/templates/_helpers.tpl]: done

File: dashbase/tests/test_0_replicas/service/searcher.yaml (hunk 1 above)
Comment (cuichenli) [reply dashbase/tests/test_0_replicas/service/searcher.yaml]: done

**Comment by cuichenli** [comment]
https://github.com/dashbase/dashbase-k8s/blob/master/dashbase/templates/_helpers.tpl#L65-L73

also i can see the usage of `default number` in those lines, not sure if it is necessary to make the change as well?

**Comment by sincejune** [comment]
> https://github.com/dashbase/dashbase-k8s/blob/master/dashbase/templates/_helpers.tpl#L65-L73
> 
> also i can see the usage of `default number` in those lines, not sure if it is necessary to make the change as well?

It's ok to leave it there. `0` is not a valid value of `minReplicas` for hpa.

**Pull request merged**